staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1395,20 +1395,22 @@ func (e *Store) finalizeDelete(ctx context.Context, obj runtime.Object, runHooks
 // automatically.
 func (e *Store) Watch(ctx context.Context, options *metainternalversion.ListOptions) (watch.Interface, error) {
 	label := labels.Everything()
-	if options != nil && options.LabelSelector != nil {
+	if options == nil {
+		// By default we should serve the request from etcd.
+		options = &metainternalversion.ListOptions{ResourceVersion: ""}
```

Comment:
I'm not sure this default is correct... I would expect `watch: true` at the very least.

I *think* there is no non-test path that calls this with nil. This is the only call I can find:

https://github.com/kubernetes/kubernetes/blob/856475e5fffe3d99c71606d6024f5ed93e37eebc/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go#L264

Should we consider changing this to a non-pointer or erroring if options is nil, rather than defaulting?

Reply:
if it's not possible for any known upper layer to call with nil today, avoiding that case by treating it as an error seems preferable to trying to pick a default that may or may not match the caller's expectations

Reply:
Thanks for review both of you. 
 
@liggitt, I agree that returning an error if options is nil provides stronger guarantees and avoids ambiguity, ensuring long-term maintainability so I'll proceed with returning an error if options is nil.

Reply:
@mouuii , I see the value in defaulting values for easier API usage, but to reduce ambiguity, I'll proceed with returning an error for nil options. 

If, in the future, we encounter a scenario where it's common for options to be nil, we can revisit the possibility of adding defaults.